src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1633,7 +1585,10 @@ void Lowering::LowerHWIntrinsicCreate(GenTreeHWIntrinsic* node)
             BlockRange().InsertAfter(tmp3, idx);
 
             node->ResetHWIntrinsicId(NI_AVX_InsertVector128, comp, tmp3, tmp1, idx);
-            return;
+            tmp1->ClearUnusedValue();
+
+            LowerNode(node);
```

Comment:
Why was `LowerNode` not needed previously?

Reply:
Because previously `LowerHWIntrinsic` called `LowerHWIntrinsicCreate` and then had its own check that called `LowerNode` again under certain circumstances. The logic was "fixed" so that `LowerHWIntrinsicCreate` does it itself and returns the next node, which simplifies things overall.